ast/src/parsed/mod.rs

```diff
@@ -151,6 +152,13 @@ impl PilStatement {
                         .map(move |v| (name, Some(&v.name), SymbolCategory::TypeConstructor)),
                 ),
             ),
+            PilStatement::StructDeclaration(_, StructDeclaration { name, fields, .. }) => Box::new(
+                once((name, None, SymbolCategory::Type)).chain(
+                    fields
```

Comment:
I still don't think the fields should be externally accessible like this. If you return them here, then something like `StructName::field_name` is valid, but I don't think it should.